workspace/server.js

```diff
@@ -1,3 +1,21 @@
 import ultra from "../server.ts";
 
-await ultra();
+await ultra({
+  middleware: [
+    async (request, next) => {
+      console.log("middleware 1 before request", request.url);
+
+      await next();
+
+      console.log("middleware 1 after request", request.url);
+    },
+
+    async (request, next) => {
+      console.log("middleware 2 before request", request.url);
+
+      await next();
+
+      console.log("middleware 2 after request", request.url);
+    },
+  ],
```

Comment:
I wonder if we should go with the `use` function pattern common to web server frameworks, this will allow more flexibility.

eg. 

```js
const server = ultra();

server.use(...);
```

Reply:
Ditto to that

Reply:
Done